modules/order/tests/src/Unit/AdjustmentTest.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Drupal\Tests\commerce_order\Unit;
+
+use Drupal\commerce_order\Adjustment;
+use Drupal\Tests\UnitTestCase;
+
+/**
+ * @coversDefaultClass Drupal\commerce_order\Adjustment
+ * @group commerce
+ */
+class AdjustmentTest extends UnitTestCase {
+
+  /**
+   * Tests the constructor and definition checks.
+   *
+   * @covers ::__construct
+   */
+  public function testAdjustmentConstruct() {
+    $definition = [
+      'type' => 'discount',
+      'label' => '10% off',
+      'amount' => -1.00,
+      'currency_code' => 'USD',
+      'source_id' => 'promotion:1',
+    ];
+
+    $adjustment = new Adjustment($definition);
+
+    unset($definition['source_id']);
+
+    $this->setExpectedException(\InvalidArgumentException::class, 'Missing required property source_id');
+    $adjustment = new Adjustment($definition);
+  }
+
+  /**
+   * Tests methods on the adjustment object.
+   *
+   * @covers ::getType
+   * @covers ::getLabel
+   * @covers ::getDecimalAmount
+   * @covers ::getCurrencyCode
+   * @covers ::getSourceId
+   * @covers ::toPrice
+   */
+  public function testAdjustmentMethods() {
+    $definition = [
+      'type' => 'discount',
+      'label' => '10% off',
+      'amount' => -1.00,
+      'currency_code' => 'USD',
+      'source_id' => 'promotion:1',
+    ];
+
+    $adjustment = new Adjustment($definition);
+    $this->assertEquals('discount', $adjustment->getType());
+    $this->assertEquals('10% off', $adjustment->getLabel());
+    $this->assertEquals(-1.00, $adjustment->getDecimalAmount());
+    $this->assertEquals('USD', $adjustment->getCurrencyCode());
+    $this->assertEquals('promotion:1', $adjustment->getSourceId());
```

Comment:
It's probably enough for the source id to be just 'one', the source object type is adjustment type specific. If you have a 'discount' adjustment, you know you want to load a promotion entity.